ui-src/src/app/automated-ui-services.ts

```diff
@@ -5,22 +5,25 @@ import { HttpClient, HttpHeaders } from "@angular/common/http";
 import { ConfigurationsComponent } from "./configurations/configurations.component";
 import { Headers } from "@angular/http/src/headers";
 import { Observable } from "rxjs/Observable";
-
-const httpOptions = {
-  headers: new HttpHeaders({
-    "Content-Type": "application/json",
-    configPathDir:
-      "C:/Users/a586754/go/src/github.com/xtracdev/automated-perf-test/config/"
-  })
-};
+import { environment } from "../environments/environment.prod";
 
 @Injectable()
 export class AutomatedUIServices {
   constructor(private http: HttpClient) {}
 
-   private url = "http://localhost:9191/configs";
+  postConfig$(configData, configPath): Observable<any> {
+    let headers = new HttpHeaders();
+    headers = headers.set("Content-Type", "application/json;");
+    headers = headers.append("configPathDir", configPath);
+
+    return this.http.post(environment.apiBaseUrl + "configs", configData, {headers});
```

Comment:
Actually this is quite expensive to do. why not use template litterals https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Is this what you had in mind
var apiBaseUrl = "http://localhost:9191/";
 return this.http.put(`${apiBaseUrl}configs/${xmlFileName}`, configData, {headers});

Reply:
Yes, that's it. Only the apiBaseUrl should be enviroment.apiBaseUrl.